### ISSUE TITLE
A language specification

### ISSUE THREAD

**Issue by cstffx** [imba/imba#308]
### Description

Work in a language specification document.

### Why

Keeping up with new features and syntax changes is complicated due to lack of documentation. But everything is easier if each feature is documented in a document that contains a formal description of the language. This could also be useful for those who wish to improve or implement pending features that involve the compiler.

### Is this something you're interested in working on?
Yes

**Comment by scanf** [comment]
I love the idea but I am not sure how much overhead this could potentially add. Do you have a concrete example? Were you thinking something in the direction of GROQ [https://sanity-io.github.io/GROQ/](https://sanity-io.github.io/GROQ/)



Thanks.

**Comment by cstffx** [comment]
Yes. I was using as a reference in my mind the [ecma-262](http://www.ecma-international.org/ecma-262/) and  you proposal is very similar. Seems that we can use [spec-nd](https://spec-md.com/)

**Comment by scanf** [comment]
https://github.com/imba/imba/blob/master/src/compiler/grammar.imba1
https://github.com/imba/imba/blob/master/src/compiler/rewriter.imba1
